### PR TITLE
[FIX] website: ignore leading slashes when creating a page

### PR DESCRIPTION
**Pull request by bso-odoo**
When the name given when creating a new page starts with a slash, the werkzeug route `/website/add/<path:path>` cannot be matched. This leads to wrong navigation situatons.

This commit removes any leading slash from the entered page name.

Steps to reproduce:
- Create a page.
- Enter "/abc" as name.

=> A 404 error page was reached.

[FIX] website: do not reuse an existing view key for a page key
When generating a new page key, it was only made sure to not match
existing page keys. This leads to COW happening on existing views if the
key already existed in a view.

This commit ensures that new page keys are not existing view keys
either.

Steps to reproduce:
- Create a page named "snippets".

=> Notification was shown indicating that `website.snippets` is private.

task-3328827

### CODE REVIEW

File: addons/website/models/website.py
Comment (rdeodoo) [inline addons/website/models/website.py:408]:
```suggestion
        while self.env['ir.ui.view'].with_context(active_test=False).sudo().search([('key', '=', key_copy)] + domain_static):
```
Replacing `website.page` by `ir.ui.view` was enough no?
Seems like the first condition is a subset of second condition which will involves the same check a second time, as `key` is a property of `ir.ui.view`, not `website.page` (`inherits`)

**Review by rdeodoo — CHANGES_REQUESTED** [review]
Maybe we could have a small unit test about this route?
```py
@http.route(['/website/add', '/website/add/<path:path>'], type='http', auth="user", website=True, methods=['POST'])
def pagenew(self, path="", add_menu=False, template=False, redirect=False, **kwargs):
```
I think you can test both fixes here by calling directly this controller method within MockRequest?

Note that maybe in the first implementation of website we should have use a more explicit `key` for page's view like `website.website_page_xx`. But your fix will be enough, even in master :+1:

**Comment by bso-odoo** [comment]
@rdeodoo Suggested changes done. Test added only for the python fix.

**Comment by bso-odoo** [comment]
@rdeodoo Merge slashes are explicitly disabled in odoo by https://github.com/odoo/odoo/commit/34f086607841f59d03318675b646fca3aede2fa9. :man_facepalming:

**Review by rdeodoo — APPROVED** [review]
@robodoo r+ rebase-ff

Ty for the check about minimalist werkzeug/flask.
I guess that since the merge_slash is disabled, is probably doesn't match because the merge is done before the match, meaning that if we don't merge the slashs, the `<path>` converter doesn't catch a path starting with a leading slash (as per the regex https://github.com/pallets/werkzeug/blob/5eed9385b861a1ae53c03eaae5a8dea4480113d6/werkzeug/routing.py#L893-L903)

Then the fix is LGTM, especially since we add it back ourselves if missing from page url in python code anyway

**Pull request closed without merging**
